<?xml version="1.0" ?>
<!-- Testing a hivemind with another robot in the swarm -->

<launch>
    <arg name="interloc_refresh_rate" default="10.0"/>

        <!-- Communication Broker -->
    <rosparam command="load" file="$(find swarmus_ros_simulation)/launch/config/broker_config.yaml" />
    <node name="commBroker" pkg="hive_connect" type="communication_broker" >
        <!-- Parameter name to use to find the robot list. Should match the yaml config file -->
        <param name="configList" value="/Broker/robots"/>
    </node>

    <!-- Robot 1 -->
    <!-- Starting the nodes after arguments definition. All nodes will be in the  -->
    <group ns="agent_1">
        <!-- Bridge -->
        <node pkg="hive_mind_bridge" type="hive_mind_bridge" name="hive_mind_bridge" output="screen">
            <param name="TCP_SERVER_PORT" type="int" value="7001"/>
            <param name="moveByTopic" type="str" value="/agent_1/navigation/moveBy"/>
        </node>


        <!-- HiveConnect -->
        <node pkg="hive_connect" type="hive_connect" name="hive_connect" output="screen" >

            <!-- BSP settings -->
            <param name="spi_mock_address" value="127.0.0.1" />
            <param name="spi_mock_port" type="int" value="8001" />

            <!-- Network settings -->
            <param name="tcp_listen_port" type="int" value="9001"/>
        </node>

        <!-- Interlocalization -->
        <!--<node pkg="swarmus_ros_simulation" type="interLocalization" name="interLocalization" output="screen">
            <param name="robot_name" type="string" value="agent_$(arg agent_id)" />
        </node>-->
    </group>
    
    <!-- Robot 2 -->
    <include file="$(find swarmus_ros_simulation)/launch/reusable/simulated_agent.launch">
        <arg name="agent_id"   default="2"/>
        <arg name="interloc_refresh_rate"   default="$(arg interloc_refresh_rate)"/>
    </include>
</launch>